graph/appender/unused_service.go

```diff
@@ -19,14 +20,14 @@ func (a UnusedServiceAppender) AppendGraph(trees *[]*tree.ServiceNode, namespace
 	istioClient, err := kubernetes.NewClient()
 	checkError(err)
 
-	deployments, err := istioClient.GetDeployments(namespaceName)
+	pods, err := istioClient.GetPods(namespaceName, labels.Set{})
```

Comment:
Just an optimization, as in the buildStaticNodeList() we are just filtering by 'app' and 'version' label, perhaps we can try to pass that as a filter, like app=* or version=*. I don't know if labels.Set{} support that wildcard, but if possible, it would reduce a lot of undesired pods.

Reply:
The k8s API supports the "exists" operator on label keys [1] but I don't know if the go client supports it as well

[1] https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

Reply:
If we can optimize I think it will certainly help.   With or without filtering we may want to remove duplicates here, as multiple pods will likely exist for the same service+version.

In general, if we are going to pull all of the service pods I wonder If we could/should do more combining of appender work.

Reply:
Changed. I added a new method in our kubernetes client to retrieve pods filtering out those that don't have the app and version labels.